### ISSUE TITLE
Conda fails to resolve deps, pegs CPU at 100%, provides no useful details

### ISSUE THREAD

**Issue by gh-andre** [conda/conda#13038]
### Checklist

- [X] I added a descriptive title
- [X] I searched open reports and couldn't find a duplicate

### What happened?

When trying to resolve dependencies of a package installed via `pip`, Conda goes away for a long, long time, pegs the CPU at 100% across all cores. It fails eventually with a cryptic error that looks like this on Ubuntu 20:
```
UnsatisfiableError: The following specifications were found to be incompatible with each other:

Output in format: Requested package -> Available versionsThe following specifications were found to be incompatible with your system:

  - feature:/linux-64::__glibc==2.31=0
  - feature:|@/linux-64::__glibc==2.31=0

Your installed version is: 2.31
```

Here's how this can be reproduced. Expand this dummy project source and build it with this command from the project source:
```
pip3 install build
python -m build
```
[conda-abc.zip](https://github.com/conda/conda/files/12443874/conda-abc.zip)

A wheel file will be built - `dist/abc-0.5.0-py3-none-any.whl`. Install this file into a Conda environment as follows (adjust path for where you are installing it from):
```
conda create --name abc --yes python=3.10
conda run --name abc pip3 install --no-deps dist/abc-0.5.0-py3-none-any.whl
```
The last command will print:
```
Processing ./abc/dist/abc-0.5.0-py3-none-any.whl
Installing collected packages: abc
Successfully installed abc-0.5.0
```
`conda list` shows expected output, with just the `abc` package from `pypi`:
```
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main
_openmp_mutex             5.1                       1_gnu
abc                       0.5.0                    pypi_0    pypi
bzip2                     1.0.8                h7b6447c_0
ca-certificates           2023.05.30           h06a4308_0
ld_impl_linux-64          2.38                 h1181459_1
libffi                    3.4.4                h6a678d5_0
libgcc-ng                 11.2.0               h1234567_1
libgomp                   11.2.0               h1234567_1
libstdcxx-ng              11.2.0               h1234567_1
libuuid                   1.41.5               h5eee18b_0
ncurses                   6.4                  h6a678d5_0
openssl                   3.0.10               h7f8727e_2
pip                       23.2.1          py310h06a4308_0
python                    3.10.12              h955ad1f_0
readline                  8.2                  h5eee18b_0
setuptools                68.0.0          py310h06a4308_0
sqlite                    3.41.2               h5eee18b_0
tk                        8.6.12               h1ccaba5_0
tzdata                    2023c                h04d1e81_0
wheel                     0.38.4          py310h06a4308_0
xz                        5.4.2                h5eee18b_0
zlib                      1.2.13               h5eee18b_0
```
Now try to update dependencies for this package with this command:
```
conda update --name abc --only-deps --yes abc
```
This is where all described above happens. Trying to print verbose isn't helpful - a tedious manual search of package dependencies yields no mismatches either.

### Conda Info

```shell
active environment : base
    active env location : /home/ubuntu/miniconda3
            shell level : 1
       user config file : /home/ubuntu/.condarc
 populated config files : /home/ubuntu/.condarc
          conda version : 23.7.3
    conda-build version : not installed
         python version : 3.10.12.final.0
       virtual packages : __archspec=1=x86_64
                          __glibc=2.31=0
                          __linux=5.15.0=0
                          __unix=0=0
       base environment : /home/ubuntu/miniconda3  (writable)
      conda av data dir : /home/ubuntu/miniconda3/etc/conda
  conda av metadata url : None
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://conda.anaconda.org/intel/linux-64
                          https://conda.anaconda.org/intel/noarch
          package cache : /home/ubuntu/miniconda3/pkgs
                          /home/ubuntu/.conda/pkgs
       envs directories : /home/ubuntu/miniconda3/envs
                          /home/ubuntu/.conda/envs
               platform : linux-64
             user-agent : conda/23.7.3 requests/2.31.0 CPython/3.10.12 Linux/5.15.0-1041-aws ubuntu/20.04.5 glibc/2.31
                UID:GID : 1000:1000
             netrc file : None
           offline mode : False
```

### Conda Config

```shell
==> /home/ubuntu/.condarc <==
pip_interop_enabled: True
channels:
  - defaults
  - conda-forge
  - intel
report_errors: False
unsatisfiable_hints: True

==> cmd_line <==
debug: False
json: False
```

### Conda list

```shell
# packages in environment at /home/ubuntu/miniconda3:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main    defaults
_openmp_mutex             5.1                       1_gnu    defaults
boltons                   23.0.0          py310h06a4308_0    defaults
brotlipy                  0.7.0           py310h7f8727e_1002    defaults
bzip2                     1.0.8                h7b6447c_0    defaults
c-ares                    1.19.0               h5eee18b_0    defaults
ca-certificates           2023.05.30           h06a4308_0    defaults
certifi                   2023.7.22       py310h06a4308_0    defaults
cffi                      1.15.1          py310h5eee18b_3    defaults
charset-normalizer        2.0.4              pyhd3eb1b0_0    defaults
conda                     23.7.3          py310h06a4308_0    defaults
conda-content-trust       0.1.3           py310h06a4308_0    defaults
conda-libmamba-solver     23.5.0          py310h06a4308_0    defaults
conda-package-handling    2.2.0           py310h06a4308_0    defaults
conda-package-streaming   0.9.0           py310h06a4308_0    defaults
cryptography              39.0.1          py310h9ce1e76_2    defaults
fmt                       9.1.0                hdb19cb5_0    defaults
icu                       58.2                 he6710b0_3    defaults
idna                      3.4             py310h06a4308_0    defaults
jsonpatch                 1.32               pyhd3eb1b0_0    defaults
jsonpointer               2.1                pyhd3eb1b0_0    defaults
krb5                      1.20.1               h143b758_1    defaults
ld_impl_linux-64          2.38                 h1181459_1    defaults
libarchive                3.6.2                h6ac8c49_2    defaults
libcurl                   8.1.1                h251f7ec_2    defaults
libedit                   3.1.20221030         h5eee18b_0    defaults
libev                     4.33                 h7f8727e_1    defaults
libffi                    3.4.4                h6a678d5_0    defaults
libgcc-ng                 11.2.0               h1234567_1    defaults
libgomp                   11.2.0               h1234567_1    defaults
libmamba                  1.4.1                h2dafd23_1    defaults
libmambapy                1.4.1           py310h2dafd23_1    defaults
libnghttp2                1.52.0               h2d74bed_1    defaults
libsolv                   0.7.22               he621ea3_0    defaults
libssh2                   1.10.0               hdbd6064_2    defaults
libstdcxx-ng              11.2.0               h1234567_1    defaults
libuuid                   1.41.5               h5eee18b_0    defaults
libxml2                   2.10.4               hcbfbd50_0    defaults
lz4-c                     1.9.4                h6a678d5_0    defaults
ncurses                   6.4                  h6a678d5_0    defaults
openssl                   3.0.10               h7f8727e_1    defaults
packaging                 23.0            py310h06a4308_0    defaults
pcre2                     10.37                he7ceb23_1    defaults
pip                       23.1.2          py310h06a4308_0    defaults
pluggy                    1.0.0           py310h06a4308_1    defaults
pybind11-abi              4                    hd3eb1b0_1    defaults
pycosat                   0.6.4           py310h5eee18b_0    defaults
pycparser                 2.21               pyhd3eb1b0_0    defaults
pyopenssl                 23.2.0          py310h06a4308_0    defaults
pysocks                   1.7.1           py310h06a4308_0    defaults
python                    3.10.12              h955ad1f_0    defaults
readline                  8.2                  h5eee18b_0    defaults
reproc                    14.2.4               h295c915_1    defaults
reproc-cpp                14.2.4               h295c915_1    defaults
requests                  2.31.0          py310h06a4308_0    defaults
ruamel.yaml               0.17.21         py310h5eee18b_0    defaults
ruamel.yaml.clib          0.2.6           py310h5eee18b_1    defaults
setuptools                67.8.0          py310h06a4308_0    defaults
six                       1.16.0             pyhd3eb1b0_1    defaults
sqlite                    3.41.2               h5eee18b_0    defaults
tk                        8.6.12               h1ccaba5_0    defaults
toolz                     0.12.0          py310h06a4308_0    defaults
tqdm                      4.65.0          py310h2f386ee_0    defaults
tzdata                    2023c                h04d1e81_0    defaults
urllib3                   1.26.16         py310h06a4308_0    defaults
wheel                     0.38.4          py310h06a4308_0    defaults
xz                        5.4.2                h5eee18b_0    defaults
yaml-cpp                  0.7.0                h295c915_1    defaults
zlib                      1.2.13               h5eee18b_0    defaults
zstandard                 0.19.0          py310h5eee18b_0    defaults
zstd                      1.5.5                hc292b87_0    defaults
```

### Additional Context

pip interoperability is enabled with this command:
```
conda config --set pip_interop_enabled True
```

This used to work around Conda 22.11 or so. However, I'm not sure if it's the Conda version that broke this or some combination of packages in the channels because this started to happen when Conda 23.5.2 was working fine with this setup for a while.

**Comment by jezdez** [comment]
Could you try this with the [conda-libmamba-solver](https://conda.github.io/conda-libmamba-solver/getting-started/) enabled, please?

**Comment by gh-andre** [comment]
@jezdez Thanks for responding.

I tried `libmamba` outside of the `base` environment, as described below. If it's important that it would be in `base`, I will need more time to test because I will need to spin up a new VM and set it up. Let me know.

These are the commands I used:
```
conda create --name abc --yes python=3.10
conda install --name abc --yes conda-libmamba-solver
conda run --name abc pip3 install --no-deps abc/dist/abc-0.5.0-py3-none-any.whl
conda update --name abc --only-deps --yes --solver libmamba abc
```
The output I observed was different on the first run and subsequent runs - seems that it saves something outside the environment. Either didn't install proper dependencies.

Here's the first run output:
```
Channels:
 - defaults
 - conda-forge
 - intel
Platform: linux-64
Collecting package metadata (repodata.json): done
Solving environment: - error    libmamba Selected channel specific (or force-reinstall) job, but package is not available from channel. Solve job will fail.
error    libmamba Selected channel specific (or force-reinstall) job, but package is not available from channel. Solve job will fail.
error    libmamba Selected channel specific (or force-reinstall) job, but package is not available from channel. Solve job will fail.
error    libmamba Selected channel specific (or force-reinstall) job, but package is not available from channel. Solve job will fail.                                                                       done

## Package Plan ##

  environment location: /home/ubuntu/miniconda3/envs/abc

  added / updated specs:
    - abc


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    ruamel.yaml.clib-0.2.6     |  py310h5eee18b_1         139 KB
    ------------------------------------------------------------
                                           Total:         139 KB

The following packages will be SUPERSEDED by a higher-priority channel:

  ruamel.yaml.clib                                    intel --> pkgs/main



Downloading and Extracting Packages

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
```
On a clean subsequent run (i.e. remove environment `abc` and re-run those commands at the top of this post), this output is observed:
```
Channels:
 - defaults
 - conda-forge
 - intel
Platform: linux-64
Collecting package metadata (repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /home/ubuntu/miniconda3/envs/abc

  added / updated specs:
    - abc


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    ruamel.yaml.clib-0.2.6     |  py310h5eee18b_1         139 KB
    ------------------------------------------------------------
                                           Total:         139 KB

The following packages will be SUPERSEDED by a higher-priority channel:

  ruamel.yaml.clib                                    intel --> pkgs/main



Downloading and Extracting Packages

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
```
In either case `abc`'s dependencies are not installed. Here's the package list in the `abc` environment - most packages are from `libmamba` and `abc`'s packages, like `pymongo` are missing.
```
# packages in environment at /home/ubuntu/miniconda3/envs/abc:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main
_openmp_mutex             5.1                       1_gnu
abc                       0.5.0                    pypi_0    pypi
boltons                   23.0.0          py310h06a4308_0
brotlipy                  0.7.0           py310h7f8727e_1002
bzip2                     1.0.8                h7b6447c_0
c-ares                    1.19.0               h5eee18b_0
ca-certificates           2023.05.30           h06a4308_0
certifi                   2023.7.22       py310h06a4308_0
cffi                      1.15.1          py310h5eee18b_3
charset-normalizer        2.0.4              pyhd3eb1b0_0
conda                     23.7.3          py310h06a4308_0
conda-libmamba-solver     23.7.0          py310h06a4308_0
conda-package-handling    2.2.0           py310h06a4308_0
conda-package-streaming   0.9.0           py310h06a4308_0
cryptography              41.0.2          py310h22a60cf_0
fmt                       9.1.0                hdb19cb5_0
icu                       73.1                 h6a678d5_0
idna                      3.4             py310h06a4308_0
jsonpatch                 1.32               pyhd3eb1b0_0
jsonpointer               2.1                pyhd3eb1b0_0
krb5                      1.20.1               h143b758_1
ld_impl_linux-64          2.38                 h1181459_1
libarchive                3.6.2                h6ac8c49_2
libcurl                   8.1.1                h251f7ec_2
libedit                   3.1.20221030         h5eee18b_0
libev                     4.33                 h7f8727e_1
libffi                    3.4.4                h6a678d5_0
libgcc-ng                 11.2.0               h1234567_1
libgomp                   11.2.0               h1234567_1
libmamba                  1.4.1                h2dafd23_1
libmambapy                1.4.1           py310h2dafd23_1
libnghttp2                1.52.0               h2d74bed_1
libsolv                   0.7.22               he621ea3_0
libssh2                   1.10.0               hdbd6064_2
libstdcxx-ng              11.2.0               h1234567_1
libuuid                   1.41.5               h5eee18b_0
libxml2                   2.10.4               hf1b16e4_1
lz4-c                     1.9.4                h6a678d5_0
ncurses                   6.4                  h6a678d5_0
openssl                   3.0.10               h7f8727e_2
packaging                 23.0            py310h06a4308_0
pcre2                     10.37                he7ceb23_1
pip                       23.2.1          py310h06a4308_0
pluggy                    1.0.0           py310h06a4308_1
pybind11-abi              4                    hd3eb1b0_1
pycosat                   0.6.4           py310h5eee18b_0
pycparser                 2.21               pyhd3eb1b0_0
pyopenssl                 23.2.0          py310h06a4308_0
pysocks                   1.7.1           py310h06a4308_0
python                    3.10.12              h955ad1f_0
readline                  8.2                  h5eee18b_0
reproc                    14.2.4               h295c915_1
reproc-cpp                14.2.4               h295c915_1
requests                  2.31.0          py310h06a4308_0
ruamel.yaml               0.17.21         py310h5eee18b_0
ruamel.yaml.clib          0.2.6           py310h5eee18b_1
setuptools                68.0.0          py310h06a4308_0
sqlite                    3.41.2               h5eee18b_0
tk                        8.6.12               h1ccaba5_0
toolz                     0.12.0          py310h06a4308_0
tqdm                      4.65.0          py310h2f386ee_0
tzdata                    2023c                h04d1e81_0
urllib3                   1.26.16         py310h06a4308_0
wheel                     0.38.4          py310h06a4308_0
xz                        5.4.2                h5eee18b_0
yaml-cpp                  0.7.0                h295c915_1
zlib                      1.2.13               h5eee18b_0
zstandard                 0.19.0          py310h5eee18b_0
zstd                      1.5.5                hc292b87_0
```
The way it used to work is that it would download all `abc`'s dependencies from Conda channels, so other Python apps using `abc` can be installed on top of it all via Conda.

**Comment by jezdez** [comment]
@jaimergp Do you happen to know if that's an expected behavior with the libmamba solver?

**Comment by dcedstrom** [comment]
Had the same issue with conda being unable to resolve packages, let it run for about 15 minutes and it just spun with CPU maxed. If there were resolution issues with the environment import it threw an exception pretty quick about not being able to find dependency versions, issue only occurred once it was trying to actually grab and install the packages. 

Found the blog post about the mamba solver and after swapping to that it worked almost instantly. All of the underlying dependencies resolved without being explicitly referenced. 

Using latest stable version with Ubuntu 22.04 on WSL if it matters.

**Comment by gh-andre** [comment]
A couple of days ago I upgraded Conda to v23.10.0 and installed `libmamba` into `base`. Since then I didn't see the errors I described in the original post. Not trying to say that this is what resolved the original problem, though. Seems to me that there are two problems here.

  * Some particular package configuration and, perhaps, differences between Conda and PyPi repositories (e.g. `pyyaml` was available as 6.01 in PyPi and only as 6.0 in Conda for a while, both match 6.0.*) were leading to those failures to solve environments, and with quite cryptic errors.
  * Conda classic solver is incredibly slow and memory-inefficient. It needs at least 8GB of RAM to solve most environments, will work with small environments on 4GB VMs and will be guaranteed to peg CPU at 100% and eventually fail on 2GB VMs.

The first problem, which triggered this issue, is no longer reproducible in my environment, so if you want to close this issue, it sounds reasonable.

The second one can be worked around with `libmamba`, at least in latest Conda versions. Maybe it should become the default solver for Conda. Otherwise, perhaps profiling memory and CPU usage of the classic solver is worth a try - C/C++ vs. Python wouldn't explain memory usage jumping gigs in seconds.

**Comment by jezdez** [comment]
Thank you, @gh-andre for the update!

> Maybe it should become the default solver for Conda.

`*steps out of time machine*` With 23.10.0 it _has become the default solver_. Here's the [release blog post](https://conda.org/blog/2023-11-06-conda-23-10-0-release).

I'm closing this as essentially fixed by using libmamba going forward.

**Comment by gh-andre** [comment]
@jezdez Question about Conda 23.10.0 and Python 3.10. I typically set up VMs with Conda using the latest package for the given Python version from this page:

https://docs.conda.io/projects/miniconda/en/latest/miniconda-other-installer-links.html#linux-installers

In my post above, I updated Conda via `conda update`, which got me to v23.10.0 on that VM, but when I started updating my scripts, I realized that the page above lists only Conda v23.9.0 for Python 3.10. Will this page be updated to include Conda v23.10.0 for Python 3.10 or this is considered an unsupported combination?

Thanks
